var/spack/repos/builtin/packages/emacs/package.py

```diff
@@ -75,64 +96,46 @@ class Emacs(AutotoolsPackage, GNUMirrorPackage):
         depends_on("libxaw", when="toolkit=athena")
         depends_on("gtkplus", when="toolkit=gtk")
 
-    # the following dependencies are required when building from a git ref
-    # so that we can run reconfigure to generate a ./configure script
-    depends_on("m4", type="build", when="@master:")
-    depends_on("autoconf", type="build", when="@master:")
-    depends_on("automake", type="build", when="@master:")
-    depends_on("libtool", type="build", when="@master:")
-
+    # Platform-specific conflicts
+    conflicts("gui=cocoa", when="platform=linux", msg="Use gui=x11 for Linux GUI support")
```

Comment:
Replace two conflicts with one requires:
```suggestion
    requires("platform=macos", when="gui=cocoa", msg="gui=cocoa only available on macOS")
```

Reply:
Normally I'd agree with moving to a single `requires()` instead of two `conflicts()` directives. However, in this case I actually like the two directives since this give users a better error message with the way to correct it.

I think most users aren't going to understand the difference between cocoa as macOS and x11 as linux without being reminded of the proper systems.

Reply:
Sounds good.